### DISCUSSION TITLE
app/Http/RedirectIfAuthenticated.php - Middleware return type issue

### DISCUSSION THREAD

**Discussion by amrudin-shopcircle** [laravel/framework#50369] [Show and tell] [answered]
Hi. I had encountered an issue with return types in my default fresh installed Laravel 10.10 application. The issue has occrued during static analysis using PHPStan package while running `../vendor/bin/phpstan analyze --level=max app/Http`.

Here is the issue that I got in a console:

` ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  Line   Middleware/RedirectIfAuthenticated.php                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  24     Method App\Http\Middleware\RedirectIfAuthenticated::handle() should return Symfony\Component\HttpFoundation\Response but returns  
         Illuminate\Http\RedirectResponse|Illuminate\Routing\Redirector.                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
`

**Comment by richt222** [comment, marked as answer]
Hey @amrudin-shopcircle ,

You need to tell PHPStan that the `handle()` method may return a redirect. Something like...

```
    /**
     * Handle an incoming request.
     *
     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Http\Response
     */
    public function handle(Request $request, Closure $next)
    {
        /* ... */
        return $next($request);
    }
```
